### ISSUE TITLE
[IDEA] Add Option to "Chat with Document(s)"

### ISSUE THREAD

**Issue by SamuelMiller** [ztjhz/FreeChatGPT#72]
My proposal is to enable users to upload a document (such as PDF, TXT, DOCX, Markdown, EPUB. PDF likely being the most common) at the beginning of a chat thread within FreeChatGPT. This would allow them to discuss and interact with the document during their chat. Examples of similar features can be found on websites like ChatPDF.com and Humata.ai.

If the cost of additional tokens is too high to implement this feature in the free public API, it could be limited to private and shared APIs.

In Discord, I shared a few tutorials on how to use text embedding to search and interact with lengthy documents. As I am not a programmer, I am unsure which approach would be more suited for FreeChatGPT. Perhaps, there are more advanced approaches since the #1 tutorial came out, making the #2 tutorial more pertinent.

1. ChatGPT With Your Docs | Full Tutorial WITH Code Examples
https://www.youtube.com/watch?v=kM3DPWO7YV4
2. Chat with Document(s) using OpenAI ChatGPT API and Text Embedding
https://blog.devgenius.io/chat-with-document-s-using-openai-chatgpt-api-and-text-embedding-6a0ce3dc8bc8

**Comment by saliksik** [comment]
I believe that it may be out of scope for the current focus of the project, which is to provide an easy-to-use chat platform with AI-powered capabilities that do not require technical expertise.

Your proposed feature of allowing users to upload documents into chat threads involves technical skills and a vector database, which would increase the complexity and cost of the project. While the use of text embedding to search and interact with lengthy documents is an interesting concept, it may require a more advanced approach that is beyond the current scope of the project.

As a platform that is geared toward front-end development and easy deployment on different hosting providers, the web app aims to make AI technology accessible to everyone without requiring specialized technical knowledge. 

I would like to suggest that you utilize the GitHub search feature to look for open source repositories that may provide a solution to your proposed feature request. There are many repositories on GitHub related to chatbots and conversational AI, and it is possible that some of them have implemented similar document upload features.

By searching GitHub, you can explore existing projects and potentially find a solution that fits your needs. You can also contribute to existing projects by opening issues or pull requests, or even building on top of them to create new features.

**Comment by SamuelMiller** [comment]
Good point about keeping the project easy to deploy on different hosting providers, without having to have a lot of technical expertise.

**Comment by SamuelMiller** [comment]
I wonder if Open AI's open-source retrieval plugin could be implemented into Free ChatGPT.
See https://github.com/openai/chatgpt-retrieval-plugin
